### PR TITLE
Fix serialization of blockedURI

### LINKED ISSUE #519: Report a violation assumes violation's resource is a URL
Violation's resource can be a variety of things, only one of which can be passed to the URL serializer. How do the other values manifest in `blockedURI`?

### PR DESCRIPTION
**Pull request by antosart**
Fixes #519

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1560,6 +1560,21 @@ this algorithm returns normally if compilation is allowed, and throws a
     };
   </pre>
 
+  <h3 id="obtain-violation-blocked-uri" algorithm>
+    Obtain the {{SecurityPolicyViolationEvent/blockedURI}} of a violation's |resource|
+  </h3>
+
+  Given a violation's <a>resource</a> (|resource|), this algorithm returns a
+  string or `null`, to be used as the blocked URI field for violation reports.
```
[the remark this answers is not in the document]
Comment (antosart) [reply index.bs]: You are right. Fixed.

File: index.bs [hunk 2]
```diff
@@ -1560,6 +1560,20 @@ this algorithm returns normally if compilation is allowed, and throws a
...
+
+  Given a violation's <a>resource</a> (|resource|), this algorithm returns a
+  [=string=], to be used as the blocked URI field for violation reports.
+
+  1. Assert: |resource| is a {{URL}} or a [=string=].
+
+  2. If |resource| is a {{URL}}, return the result of executing the <a>URL
+     serializer</a> on |resource|, with the `exclude fragment` flag set.
+
+  3. Else if |resource| is a [=string=], then return |resource|.
```
Comment (annevk) [inline index.bs:1575]: "Return |resource|" seems better here given the assert above.

**Review by annevk — COMMENTED** [review]
You need to update https://w3c.github.io/webappsec-csp/#violation-resource as well to make it clear what the string for null is. And it seems that needs to be changed in other places too.

**Comment by annevk** [comment]
This new commit only fixes the nit, not the larger comment of what null means. Perhaps that just needs to be rewritten as "`null`" in various places?

**Comment by antosart** [comment]
I think violation's resource can be null only "temporarily" (while it is being constructed), but by the time a violation is reported (and hence the blockedURI is serialized), violation's resource should always be filled. I followed the call sites of the algorithm in the spec and that seems to be the case.

Are you just writing that we should somehow explain this in https://w3c.github.io/webappsec-csp/#violation-resource?

**Comment by annevk** [comment]
Ah okay, https://w3c.github.io/webappsec-csp/#create-violation-for-global confused me. It might be nice to point out that it's only a temporary value, but not needed as part of this PR.

**Review by annevk — APPROVED** [review]
Thanks!

**Pull request merged**